src/Utilities/functions.jl

```diff
@@ -378,11 +377,16 @@ function Base.getindex(
     I::AbstractVector,
 ) where {T}
     terms = MOI.VectorAffineTerm{T}[]
+    # assume at least one term per index
+    sizehint!(terms, length(I))
     constant = Vector{T}(undef, length(I))
     for (i, j) in enumerate(I)
-        g = it[j]
-        append!(terms, map(t -> MOI.VectorAffineTerm(i, t), g.terms))
-        constant[i] = g.constant
+        for term in it.f.terms
+            if term.output_index == j
```

Comment:
We could replace this by `term.output_index in I` and remove the outer loop. That would be more efficient if `I` is a range

Reply:
yup, just did in the last commit